### PR TITLE
Monitoring infrastructure

### PR DESCRIPTION
**Pull request by leosoto**
First stab at the way we will monitor services. It's not fully tested yet, but it should provide a good foundation once we get to the stories where we monitor and show the state of services.

For simplicity everything is mostly handled by the `ServiceVersion` class (we could extract it to a module later if we think the model is getting too fat):

- `#schedule_health_checks` will be invoked after any modification of a service version and will take care of scheduling or removing health checks depending on the state of the service version.

- `#scheduled_health_check_frequency` is going to be modified later to read some configuration (dunno if it's going to be per-organization or per-service) to figure out how frequently should the service be monitored. By now it defaults to 1-minute frequency for everything.

- Via `Sidekiq::Cron` the `ServiceVersionMonitorWorker` gets scheduled on the desired frequency

- ...and it calls back to `ServiceVersion#perform_health_check` which is the straightforward monitoring code.

- Whatever the result of the monitoring, an entry is added to the `service_version_health_checks` table, which is going to be — in practice — a huuuge log of monitoring results. A quick calculation says that if we the services we monitor are pinged on average every 3 minutes we get 60/3 * 20 *24  = 480 new daily records *per service*. With 1000 services in the platform, we get 480.000 records per day which means ~200 million records per year. With proper indexing  — not included yet — PostgreSQL should be able to handle that. And when/if it's not, we should be able to prune/archive old logs easily.

So, I think this is a viable strategy for our monitoring infrastructure :)

### CODE REVIEW

File: app/models/service_version.rb [hunk 1]
```diff
@@ -327,4 +329,71 @@ def _resolve_path(original_path, path_params)
...
+  def health_check_response
+    RestClient.get(monitor_url)
+  rescue RestClient::Exception => e
+    e.response
+  end
+
+  def perform_health_check!
+    response = health_check_response
+    plain_body = response.body
+    json_body = JSON.parse(plain_body) rescue nil
```
Comment (luelher) [inline app/models/service_version.rb]: there is not necessary to check the format of the response??

Comment (leosoto) [reply app/models/service_version.rb]: Yep: it's gonna be nil if the `plain_body` isn't parseable. Couple of lines below that case is handled and in `status_message` we store `"Not a JSON response: #{plain_body.inspect}"`

Comment (luelher) [reply app/models/service_version.rb]: Sorry @leosoto I don't mean that, I'm talking about the format exposed in the interoperability standard doc.

`
{
“codigo_estado”: 503
“msj_estado”: “Service Unavailable”
“desc_personalizada_estado”: “Servicio en mantenimiento”
}
`
The response could be a valid JSON, but not necessary has the required format (keys)

Comment (leosoto) [reply app/models/service_version.rb]: Oh, you are right — I guess we should validate that at least those 3 keys are present in the JSON response!

Comment (leosoto) [reply app/models/service_version.rb]: @luelher Just updated the pull request with the validation

**Pull request merged**